### PR TITLE
Limit the number of progressive scans for jpeg decoding

### PR DESCRIPTION
**Pull request by stiepan**
## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
Malicious jpeg file can cause the libjpeg-turbo to spin for hours in the huge number of progressive scans. There's no clear distinction in the format on when that's desired or malicious behaviour, so we are setting a limit configurable with env variable.

## Additional information:

https://www.libjpeg-turbo.org/pmwiki/uploads/About/TwoIssueswiththeJPEGStandard.pdf

### Affected modules and functionalities:
Image jpeg decoder.

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [x] Affects existing requirements (SEC.04)
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-3618

### CODE REVIEW

File: dali/imgcodec/decoders/jpeg/jpeg_handle.cc [hunk 1]
```diff
@@ -36,6 +36,48 @@ void CatchError(j_common_ptr cinfo) {
   longjmp(*jpeg_jmpbuf, 1);
 }
 
+inline int GetLibJpegMaxProgressiveScansEnv() {
+  static constexpr int kDefaultScansNum = 500;
+  char *env = getenv("DALI_MAX_JPEG_SCANS");
```
Comment (jantonguirao) [inline dali/imgcodec/decoders/jpeg/jpeg_handle.cc:41]: that's a good start. I would consider adding an operator parameter, but it can be added later.

**Review by jantonguirao — APPROVED** [review]

File: dali/imgcodec/decoders/jpeg/jpeg_handle.cc (hunk 1 above)
Comment (stiepan) [reply dali/imgcodec/decoders/jpeg/jpeg_handle.cc:41]: I though about exposing it as an operator parameter, but in this particular case, I believe higher values (even this one) are unlikely. Reportedly, typical values are up to 10.

Comment (stiepan) [reply dali/imgcodec/decoders/jpeg/jpeg_handle.cc:41]: And I took 500 from libjpeg-turbo TJ API - there, if you specify a boolean flag to limit the scans they pick 500. Libtiff, on the other hand, goes for 100. In any case, I don't expect users to really adjust that.

**Comment by stiepan** [comment]
!build

**Review by mzient — APPROVED** [review]

File: dali/imgcodec/decoders/jpeg/jpeg_handle.cc (hunk 1 above)
Comment (stiepan) [reply dali/imgcodec/decoders/jpeg/jpeg_handle.cc:41]: 256 sounds like a good candidate then.

**Comment by stiepan** [comment]
!build

**Pull request merged**